homeworks/10/Chat-room/src/Server.java

```diff
@@ -0,0 +1,152 @@
+                if(conn.sender == true){
+                    continue;
+                }
+                conn.sendMessage(message);
+            }
+
+        }
+    }
+
+    public void close(){
+        done = true;
+        if (!server.isClosed()){
+            try {
+                server.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        for(ConnectionHandler ch: connectionHandlers){
+            ch.close();
+        }
+    }
+
+
+    class ConnectionHandler implements Runnable{
+
+        private Socket client;
+        private BufferedReader in;
+        private PrintWriter out;
+        private String nickname;
+        private boolean sender;
+
+        public ConnectionHandler(Socket client) {
+            this.client = client;
+            sender = false;
+        }
+
+        @Override
+        public void run() {
+            try{
+                out = new PrintWriter(client.getOutputStream(), true);
+                in = new BufferedReader(new InputStreamReader(client.getInputStream()));
+                out.println("Welcome!\nYou can use the following commands\n" +
+                        "\t /nick to change your username\n" +
+                        "\t /msg to send a message to other users\n" +
+                        "\t /quit to exit\n");
+                nickname = "User";
+                System.out.println("New user has connected!");
+                broadcastMessage("New user joined the chat!");
+
+                String message;
+
+                while ((message = in.readLine()) != null){
+                    if(message.startsWith("/nick")){
+                        String[] newMessage = message.split(" ");
+                            broadcastMessage(nickname + " changed his nickname to " + newMessage[1]);
+                            nickname = newMessage[1];
+                    }else if (message.startsWith("/quit")) {
+                        SimpleDateFormat formatDate = new SimpleDateFormat("hh:mm");
+                        String time = formatDate.format(new Date());
+                        sender = true;
+                        broadcastMessage("[" + time + "] " + nickname + " left the chat!");
+                        sender = false;
+                        System.out.println(nickname + " left the chat");
+                        close();
+                    }else if (message.startsWith("/msg")){
+                        SimpleDateFormat formatDate = new SimpleDateFormat("hh:mm");
+                        String time = formatDate.format(new Date());
+                        String[] newMessage = message.split(" ", 2);
+                        sender = true;
+                        broadcastMessage("["+time+"] " + nickname + ": " + newMessage[1]);
+                        sender = false;
+                    }else{
+                    out.println("Wrong command!\n You can use the following commands\n" +
```

Comment:
You could use String.format()  or a single text block on multiple lines to make the code more readable.